server/src/main/java/org/elasticsearch/index/engine/InternalEngine.java

```diff
@@ -2845,6 +2848,10 @@ protected void commitIndexWriter(final IndexWriter writer, final Translog transl
         }
     }
 
+    protected Map<String, String> getCommitExtraUserData() {
```

Comment:
A quick javadoc to explain that some keys may be reserved by ES and replaced, like ES_VERSION, may be useful.